frontend/src/services/HTTPHelper.ts

```diff
@@ -0,0 +1,368 @@
+/* eslint-disable max-classes-per-file */
+import { injectable } from 'inversify';
+import 'reflect-metadata';
+import ArgumentError from '../errors/ArgumentError';
+import CancellationError from '../utils/CancellationError';
+import { CancellationToken } from '../utils/CancellationToken';
+import nop from '../utils/nop';
+import { HTTPHelperOptions } from './HTTPHelperOptions';
+import NetworkError from './NetworkError';
+
+interface HTTPHeaderCollection {
+  [key: string]: string | undefined;
+}
+
+type URLQueryValueType = number | string | (number | string)[];
+
+interface URLQuery {
+  [key: string]: URLQueryValueType;
+}
+
+export abstract class HTTPRequest {
+  // TODO: Why is this constructor useless? It protects the class to be consistent.
+  // eslint-disable-next-line no-useless-constructor
+  public constructor(
```

Comment:
try to add this to your `.eslintrc.json`
```jsonc
"rules": {
    "no-useless-constructor": "off", // Disable eslint no-useless-constructor
    "@typescript-eslint/no-useless-constructor": ["error"] // enable typescript no-useless-constructor
}
```
Tthe current eslint apparently does not know that you can put class attributes directly in the constructor.
Maybe you have to add @typescript-eslint to the package.json. See backend.